File: ML for Satyavrat/UDP_Client.py
import socket

def start_client(host='127.0.0.1', port=65432):
    client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    
    try:
        client_socket.connect((host, port))
        print(f"Connected to server at {host}:{port}")
        
        # Send and receive data
        while True:
            message = input("Enter a message to send to the server (or 'exit' to quit): ")
            if message.lower() == 'exit':
                break
            client_socket.sendall(message.encode())
            response = client_socket.recv(1024)
            print(f"Server response: {response.decode()}")
    
    except Exception as e:
        print(f"Error: {e}")
    finally:
        client_socket.close()
        print("Connection closed")

if __name__ == "__main__":
    start_client()

File: ML for Satyavrat/UDP_Client.java
import java.net.*;
import java.util.Scanner;

public class UDP_Client {
    public static void main(String[] args) {
        try (DatagramSocket socket = new DatagramSocket(); Scanner scanner = new Scanner(System.in)) {
            InetAddress server = InetAddress.getByName("localhost");
            int port = 9876;

            System.out.println("Type messages to send:");
            while (true) {
                // Send message
                byte[] sendData = scanner.nextLine().getBytes();
                socket.send(new DatagramPacket(sendData, sendData.length, server, port));

                // Receive response
                byte[] receiveData = new byte[1024];
                DatagramPacket response = new DatagramPacket(receiveData, receiveData.length);
                socket.receive(response);

                System.out.println("Server: " + new String(response.getData(), 0, response.getLength()));
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
